app/models/concerns/blacklight/solr/document/marc_export.rb

```diff
@@ -43,56 +45,45 @@ def export_as_chicago_citation_txt
 
   # Exports as an OpenURL KEV (key-encoded value) query string.
   # For use to create COinS, among other things. COinS are
-  # for Zotero, among other things. TODO: This is wierd and fragile
-  # code, it should use ruby OpenURL gem instead to work a lot
-  # more sensibly. The "format" argument was in the old marc.marc.to_zotero
-  # call, but didn't neccesarily do what it thought it did anyway. Left in
-  # for now for backwards compatibilty, but should be replaced by
-  # just ruby OpenURL.
+  # for Zotero, among other things.
+
   def export_as_openurl_ctx_kev(format = nil)
-    title = to_marc.find { |field| field.tag == '245' }
-    author = to_marc.find { |field| field.tag == '100' }
-    corp_author = to_marc.find { |field| field.tag == '110' }
-    publisher_info = to_marc.find { |field| field.tag == '260' }
-    edition = to_marc.find { |field| field.tag == '250' }
-    isbn = to_marc.find { |field| field.tag == '020' }
-    issn = to_marc.find { |field| field.tag == '022' }
-    unless format.nil?
-      format = format.is_a?(Array) ? format[0].downcase.strip : format.downcase.strip
-    end
-    export_text = ""
+    format = self[:format_ssim]
+    format = format.is_a?(Array) ? format[0].downcase.strip : format.downcase.strip if format.present?
+
+    context_object = OpenURL::ContextObject.new
 
     if format == 'book'
-      export_text += "ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Abook&amp;rfr_id=info%3Asid%2Fblacklight.rubyforge.org%3Agenerator&amp;rft.genre=book&amp;"
-      export_text += "rft.btitle=#{title.nil? || title['a'].nil? ? '' : CGI.escape(title['a'])}+#{title.nil? || title['b'].nil? ? '' : CGI.escape(title['b'])}&amp;"
-      export_text += "rft.title=#{title.nil? || title['a'].nil? ? '' : CGI.escape(title['a'])}+#{title.nil? || title['b'].nil? ? '' : CGI.escape(title['b'])}&amp;"
-      export_text += "rft.au=#{author.nil? || author['a'].nil? ? '' : CGI.escape(author['a'])}&amp;"
-      export_text += "rft.aucorp=#{CGI.escape(corp_author['a']) if corp_author['a']}+#{CGI.escape(corp_author['b']) if corp_author['b']}&amp;" if corp_author.present?
-      export_text += "rft.date=#{publisher_info.nil? || publisher_info['c'].nil? ? '' : CGI.escape(publisher_info['c'])}&amp;"
-      export_text += "rft.place=#{publisher_info.nil? || publisher_info['a'].nil? ? '' : CGI.escape(publisher_info['a'])}&amp;"
-      export_text += "rft.pub=#{publisher_info.nil? || publisher_info['b'].nil? ? '' : CGI.escape(publisher_info['b'])}&amp;"
-      export_text += "rft.edition=#{edition.nil? || edition['a'].nil? ? '' : CGI.escape(edition['a'])}&amp;"
-      export_text += "rft.isbn=#{isbn.nil? || isbn['a'].nil? ? '' : isbn['a']}"
-    elsif format.match?(/journal/i) # checking using include because institutions may use formats like Journal or Journal/Magazine
```

Comment:
@alexBLR Based on @eporter23's recommendation, I removed the journal use case, and we now only different between formats `book` and `dc`.